askama_shared/src/path.rs

```diff
@@ -5,9 +5,17 @@ use std::path::{Path, PathBuf};
 
 use toml;
 
+fn search_template_in_dirs(dirs: &Vec<PathBuf>, path: &Path) -> Option<PathBuf> {
+    dirs
+        .iter()
+        .find(|dir| dir.join(path).exists())
+        .map(|dir| dir.join(path))
+}
+
 pub fn get_template_source(tpl_path: &Path) -> String {
-    let mut path = template_dir();
-    path.push(tpl_path);
+    let dirs = template_dirs();
+    let path = search_template_in_dirs(&dirs, tpl_path)
+        .expect(&format!("Template file '{}' does not exist.", tpl_path.to_str().unwrap()));
```

Comment:
Nit: elsewhere I use sentence fragments for errors, not full sentences. Please don't capitalize, and don't end with a period.